001-clarifications-eval.txt

```diff
@@ -0,0 +1,754 @@
+    headers of all the entries in the .debug_info, .debug_addr,
+    .debug_line, .debug_rnglists, .debug_rnglists.dwo, .debug_loclists,
+    and .debug_loclists.dwo sections corresponding to any given program
+    location do not match.
+    --------------------------------------------------------------------
+
+In Section 2.5.x (originally 2.5.1) General Operations, replace the
+first paragraph:
+
+    --------------------------------------------------------------------
+    Each general operation represents a postfix operation on a simple
+    stack machine. Each element of the stack has a type and a value, and
+    can represent a value of any supported base type of the target
+    machine. Instead of a base type, elements can have a generic type,
+    which is an integral type that has the size of an address on the
+    target machine and unspecified signedness. The value on the top of
+    the stack after “executing” the DWARF expression is taken to be the
+    result (the address of the object, the value of the array bound, the
+    length of a dynamic string, the desired value itself, and so on).
+    --------------------------------------------------------------------
+
+... with the following:
+
+    --------------------------------------------------------------------
+    Each general operation represents a postfix operation on a simple
+    stack machine.
+
+    A value on the stack has a type and a literal value. It can
+    represent a literal value of any supported base type of the target
+    architecture. The base type specifies the size, encoding, and
+    endianity of the literal value.
+
+    There is a distinguished base type termed the generic type, which is
+    an integral type that has the size of an address in the target
+    architecture default address space, a target architecture defined
+    endianity, and unspecified signedness.
+
+    [non-normative] The generic type is the same as the unspecified type
+    used for stack operations defined in DWARF Version 4 and before.
+
+    An integral type is a base type that has an encoding of
+    DW_ATE_signed, DW_ATE_signed_char, DW_ATE_unsigned,
+    DW_ATE_unsigned_char, DW_ATE_boolean, or any target architecture
+    defined integral encoding in the inclusive range DW_ATE_lo_user to
+    DW_ATE_hi_user.
+    --------------------------------------------------------------------
+
+    > [For further discussion...]
+    > It may be desirable to add an implicit pointer base type encoding.
+    > It would be used for the type of the value that is produced when
+    > the DW_OP_deref* operation retrieves the full contents of an
+    > implicit pointer location storage created by the
+    > DW_OP_implicit_pointer operation. The literal value would record
+    > the debugging information entry and byte displacement specified by
+    > the associated DW_OP_implicit_pointer operation.
+    >
+    > It is unclear if DW_ATE_address is an integral type. GDB does not
```

Comment:
Looks like DWARF does, though:
~~~
1 7. DW_OP_deref
2 The DW_OP_deref operation pops the top stack entry and treats it as an
3 address. The popped value must have an integral type. The value retrieved
~~~

Reply:
It is not clear to me if the word address there is implying DW_ATE_address base type. Elsewhere it states that an address is represented by the generic type. It requires the generic type to be an integral type of unspecified signedness. It does not define if DW_ATE_address is considered an integral type.

gdb does not treat DW_ATE_address as an integral type so seems no current producer is using DW_ATE_address.

Reply:
DWARF 5 enumerates the base type encodings in Table 5.1. But it does not state which encodings are allowed for integral types. gdb does not appears to consider the  DW_ATE_address encoding as valid for integral types.